### ISSUE TITLE
[iOS] Some elements of an artboard seem to be positioned incorrectly.

### ISSUE THREAD

**Issue by yukimatsus** [rive-app/rive-flutter#346]
### Description

Some elements of an artboard seem to be positioned incorrectly.
For detail, please see "Expected behavior".

### Steps To Reproduce
1. Run the following code.
2. Click on '+' FAB.
3. As mentioned above, animation does not seem to work correctly.

<details>
<summary>code:</summary>

```main.dart
import 'package:flutter/material.dart';
import 'package:rive/rive.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    final type = _counter % 2;
    return Scaffold(
      backgroundColor: const Color(0xFF000000),
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: switch (type) {
          1 => Stack(
            clipBehavior: Clip.none,
            children: [
              const SizedBox(width: 120, height: 120),
              Positioned(
                top: -(1400 - 120) / 2,
                left: -(1400 - 120) / 2,
                right: -(1400 - 120) / 2,
                bottom: -(1400 - 120) / 2,
                child: RiveAnimation.asset(
                  'assets/animations/result.riv',
                ),
              ),
            ],
          ),
          _ => const SizedBox(width: 120, height: 120),
        },
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ),
    );
  }
}

```

</details>

#### Source `.riv`/`.rev` file

[result_riv_and_result_rev.zip](https://github.com/rive-app/rive-flutter/files/12658428/result_riv_and_result_rev.zip)

### Expected behavior

|-|Actual(iOS)|Expected1(Android)|Expected2(Preview)|
|-|-|-|-|
|GIF|![unexpected](https://github.com/rive-app/rive-flutter/assets/56015262/55a14ce1-76e3-4946-9358-561688e1f293)|![expected1](https://github.com/rive-app/rive-flutter/assets/56015262/1f30c2ff-8826-4786-bd95-5c280c24304f)|![expected2](https://github.com/rive-app/rive-flutter/assets/56015262/92d3880d-a3bb-46f3-8897-ed1bb9eda9f0)|
|Movie|https://github.com/rive-app/rive-flutter/assets/56015262/e801bd89-203a-4b5f-8495-483ed327b096|https://github.com/rive-app/rive-flutter/assets/56015262/7ca0a20c-9750-4c6d-a3ec-e87f87daa9cf|-|
|Description|![ios_screenshot](https://github.com/rive-app/rive-flutter/assets/56015262/f763a73f-1475-486e-a682-2c307c00b603)<br>・The nested artboard "m_lightC-scale" scales from left-top of the artboard. Expected to scale from center.<br>・The image "shine.png" is positioned at left-top of the artboard. Expected to be positioned at center.|result.rive works as expected when using rive-flutter in Android.<br>・The nested artboard "m_lightC-scale" scales not from left-top but from center of the artboard.<br>・The image "shine.png" is positioned at not left-top but center of the artboard.|Like in android, In [preview](https://rive.app/preview/), result.riv works as expected.|

### Device & Versions (please complete the following information)
- iPhone 8 / iOS 16.6.1
- Simulator(iPhone14 / iOS 16.4)

```
% flutter --version
Flutter 3.10.4 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 682aa387cf (4 months ago) • 2023-06-05 18:04:56 -0500
Engine • revision 2a3401c9bb
Tools • Dart 3.0.3 • DevTools 2.23.1

% flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.4, on macOS 13.3.1 22E772610a darwin-x64, locale ja-JP)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] Connected device (2 available)
[✓] Network resources

• No issues found!

```

Thank you for reading.

**Comment by HayesGordon** [comment]
Hi @yukimatsus, before I take a deeper look it might be an inconsistency between Impeller and Skia. Newer versions of Flutter use Impeller by default for iOS, but Android is still using Skia.

Can you confirm if you're seeing the same behaviour on iOS when you disable Impeller:
`flutter run --no-enable-impeller`

You can read more about Impeller here: flutter run --no-enable-impeller

**Comment by yukimatsus** [comment]
Hi @HayesGordon, thank you for looking into this.
With `--no-enable-impeller`, the animation works as expected.

**Comment by suryansh-t** [comment]
Hi, facing the same issue. Is there any workaround for this?

**Comment by agmyryak** [comment]
Faced same issue - quite critical one it force app to disable impeller in IOS to make it work
Is there any plans to fix it?

**Comment by HayesGordon** [comment]
Hi all,

Over the last few months there have been many similar reports that detail inconsistencies between Rive on Skia and Impeller. All of the past reports have been fixed, and the one mentioned here I believe to be resolved (from what I can tell).

https://github.com/rive-app/rive-flutter/assets/13705472/897511f6-ab04-46e0-9cfd-d77d4d5d681f

I tested this on Flutter v3.19.0

Please note that this is not controllable from Rive's side. However, we can assist with filing an issue on the Flutter channel if you can share an example .riv (ideally not complex to make it easier for the Flutter team to investigate).

You can find more information in our doc here: https://rive.app/community/doc/overview/docD20dU9Rod#note-on-rendering-in-flutter

One important thing to try before filing an issue is to verify if it may already be solved. Try running the Flutter app on the latest master channel. It is possible that visual bugs may be resolved on the latest Flutter commits, but not yet released in the beta or stable channel.

**Comment by HayesGordon** [comment]
@agmyryak can you share a new example (.riv, .rev, project) that reproduces the issue?

**Comment by yukimatsus** [comment]
@HayesGordon 
The issue I posted resolved in Flutter 3.19.0. Thank you very much.

**Comment by HayesGordon** [comment]
Thanks for confirming @yukimatsus, I'll go ahead and close this issue.

If anyone has encounters additional problems with Impeller please reopen or create a new issue and provide a sample.
